File: code/exxons/palindrome_anagram.py
print('\n')
user_word = input('which word would you like to check? ')
print('\n')
reversed_word_list = []
ordered_word_list = []



def check_anagram(user_word): 
    lower_case = (user_word.lower())
    #return (lower_case)
    
    no_space_string = (lower_case.replace(' ',''))
    
    
    non_arranged_word = (no_space_string)
    arranged_word = sorted(non_arranged_word)
    return ordered_word_list.append(arranged_word)
    #ordered_word_list.append(arranged_word)


(check_anagram(user_word))


def check_palindrome(user_word):
    user_word = user_word[::-1]
    reversed_word_list.append(user_word)
    return (user_word)




#user_word = user_word


(check_palindrome(user_word))


if user_word == reversed_word_list[0]:
    print(f'{user_word} is a palindrome!')

if user_word != reversed_word_list[0]:
        print(f'{user_word} is not a palindrome')   

    
    
    
print('\n\n\t--- Now to see if you can find an anagram for your word ---\n\n')
# print('\n')
# print('\n'
print(f'your original word was \'{user_word}\'')

(check_anagram(user_word))
print('\n')




user_word2 = input('Enter your second word: ' )
print('\n')

(check_anagram(user_word2))
    

if ordered_word_list[0] == ordered_word_list[2]:
    print(f'yes \'{user_word}\' and \'{user_word2}\' are anagrams')
    
if ordered_word_list[0] != ordered_word_list[2]:
    print(f'no \'{user_word}\' and \'{user_word2}\' are  not anagrams')

File: code/exxons/JavaScript/lab_2a/palindrome_anagram.js


let user_word = prompt("which word would you like to check?");







function check_anagram () {
    lower_case = `${user_word2}`.toLowerCase();
    
    no_space_string = lower_case.replace(/\s/g,'');

    
    non_arranged_word = no_space_string;

    no_space_string = non_arranged_word.split('').sort().join('');

    arranged_word = no_space_string;
    
    
    console.log(no_space_string);

    if (lower_case === user_word2) {
        alert(`${lower_case} and ${user_word2} are anagrams!`);
    } else {
        alert(`${lower_case} and ${user_word2} are not anagrams!`);
    }


}





function check_palindrome() {

const str = user_word

lower = str.toLowerCase();

const userPalword = lower 
    .split('')
    .reverse()
    .join('');
    
    console.log(userPalword);
    

    if (user_word === userPalword){
        alert(`The word is a palindrome`);
    } else {    alert('Sorry, thats not a palindrome.');
};


}








check_palindrome(); 


let user_word2 = prompt("now enter another word to see if it is an anagram for your word:");


check_anagram()